### PR TITLE
Feature/sg 216 remove dataset interface

### PR DESCRIPTION
**Pull request by shaydeci**

### CODE REVIEW

File: tests/unit_tests/strictload_enum_test.py [hunk 1]
```diff
@@ -54,7 +54,7 @@ def setUpClass(cls):
 
         # Save the model's state_dict checkpoint in Trainer format
         cls.trainer = Trainer("load_checkpoint_test", model_checkpoints_location='local')  # Saves in /checkpoints
-        cls.trainer.build_model(cls.original_torch_model, arch_params={'num_classes': 10})
+        cls.trainer.set_net(cls.original_torch_model)
```
Comment (ofrimasad) [inline tests/unit_tests/strictload_enum_test.py:57]: what is this?

**Review by ofrimasad — COMMENTED** [review]
Very nice work! 

in the example code, please use `dataloaders.dataset_ame(...)`. this makes it much clearer 
some minor comments inline

File: src/super_gradients/recipes/dataset_params/cifar10_dataset_params.yaml [hunk 2]
```diff
@@ -4,7 +4,7 @@ val_batch_size: 512 # batch size for valset in DatasetInterface
 # TODO: REMOVE ABOVE, HERE FOR COMPATIBILITY UNTIL WE REMOVE DATASET_INTERFACE
 
 train_dataset_params:
-  root: /data/cifar10
+  root: ./data/cifar10
```
[the remark this answers is not in the document]
Comment (shaydeci) [reply src/super_gradients/recipes/dataset_params/cifar10_dataset_params.yaml:7]: /data was inaccessible for download in our servers, this was also the data root in dataset interface.
We can switch to /data but note that on our servers (tzags) downloading the dataset will fail.

File: tests/unit_tests/strictload_enum_test.py (hunk 1 above)
Comment (shaydeci) [reply tests/unit_tests/strictload_enum_test.py:57]: we nee to simulate checkpoint saving that occurs during training, without changing the models weights so we can compare them.
Beforehand, we could just pass the net to build_model, and call save_checkpoint. Now, we have to set the net manually before calling it.

File: src/super_gradients/recipes/dataset_params/cifar10_dataset_params.yaml [hunk 3]
```diff
@@ -31,7 +31,7 @@ train_dataloader_params:
   pin_memory: True
 
 val_dataset_params:
-  root: /data/cifar10
+  root: ./data/cifar10
```
[the remark this answers is not in the document]
Comment (shaydeci) [reply src/super_gradients/recipes/dataset_params/cifar10_dataset_params.yaml:34]: See response above.

File: src/super_gradients/recipes/dataset_params/coco_detection_dataset_params.yaml [hunk 4]
```diff
@@ -1,81 +1,3 @@
-data_dir: /data/coco # root path to coco data
```
[the remark this answers is not in the document]
Comment (shaydeci) [reply src/super_gradients/recipes/dataset_params/coco_detection_dataset_params.yaml:1]: Not quite. It has a lot of resemblance but with suffixes that are later mapped by the dataset interface, where in the new format we divide the parameters between train_dataset_params and val_dataset_params (and inside them, the argument names are the same)

File: src/super_gradients/examples/ddrnet_imagenet/ddrnet_classification_example.py [hunk 5]
```diff
@@ -56,19 +56,25 @@
...
+                    ColorJitter(0.4, 0.4, 0.4),
+                    ToTensor(),
+                    Normalize(mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225]),
+                    RandomErase(0.2, "random")
+                    ]
+
 trainer = Trainer(experiment_name=args.experiment_name,
                   multi_gpu=MultiGPUMode.DISTRIBUTED_DATA_PARALLEL if distributed else MultiGPUMode.DATA_PARALLEL,
                   device='cuda')
+train_loader = imagenet_train(dataset_params={"transforms": train_transforms}, dataloader_params={"batch_size": args.batch})
```
[the remark this answers is not in the document]
Comment (shaydeci) [reply src/super_gradients/examples/ddrnet_imagenet/ddrnet_classification_example.py]: Agree, changed in https://github.com/Deci-AI/super-gradients/pull/356/commits/305ecfd20c3e3606bfe17d72d936fa0552c90417

File: src/super_gradients/examples/early_stop/early_stop_example.py [hunk 6]
```diff
@@ -22,10 +24,7 @@
...
-# Connect Dataset
-dataset = Cifar10DatasetInterface()
-trainer.connect_dataset_interface(dataset, data_loader_num_workers=8)
-
 # Build Model
-trainer.build_model("resnet18_cifar")
-trainer.train(training_params=train_params)
+model = models.get("resnet18_cifar", num_classes=10)
+
+trainer.train(model=model, training_params=train_params, train_loader=cifar10_train(), valid_loader=cifar10_val())
```
[the remark this answers is not in the document]
Comment (shaydeci) [reply src/super_gradients/examples/early_stop/early_stop_example.py]: Changed in https://github.com/Deci-AI/super-gradients/commit/305ecfd20c3e3606bfe17d72d936fa0552c90417

File: src/super_gradients/examples/regseg_transfer_learning_example/regseg_transfer_learning_example.py [hunk 7]
```diff
@@ -1,32 +1,30 @@
...
     PadShortToCropSize, ColorJitterSeg
 from super_gradients.training.utils.callbacks import BinarySegmentationVisualizationCallback, Phase
-from torchvision import transforms
-
+from super_gradients.training.dataloaders.dataloaders import supervisely_persons_val, supervisely_persons_train
 # DEFINE DATA TRANSFORMATIONS
-dataset_params = {
-    "image_mask_transforms_aug": transforms.Compose([ColorJitterSeg(brightness=0.5, contrast=0.5, saturation=0.5),
+
+dl_train = supervisely_persons_train(dataset_params={"transforms": [ColorJitterSeg(brightness=0.5, contrast=0.5, saturation=0.5),
```
[the remark this answers is not in the document]
Comment (shaydeci) [reply src/super_gradients/examples/regseg_transfer_learning_example/regseg_transfer_learning_example.py]: Changed in https://github.com/Deci-AI/super-gradients/commit/305ecfd20c3e3606bfe17d72d936fa0552c90417

File: src/super_gradients/examples/resnet_qat/resnet_qat_example.py [hunk 8]
```diff
@@ -16,23 +16,23 @@
...
 super_gradients.init_trainer()
 
-dataset = ImageNetDatasetInterface(data_dir="/data/Imagenet", dataset_params={"batch_size": 128})
 trainer = Trainer("resnet18_qat_example",
                   model_checkpoints_location='local',
                   multi_gpu=MultiGPUMode.DISTRIBUTED_DATA_PARALLEL)
 
-trainer.connect_dataset_interface(dataset)
-trainer.build_model("resnet18", checkpoint_params={"pretrained_weights": "imagenet"})
+train_loader = imagenet_train()
```
[the remark this answers is not in the document]
Comment (shaydeci) [reply src/super_gradients/examples/resnet_qat/resnet_qat_example.py]: Changed in https://github.com/Deci-AI/super-gradients/commit/305ecfd20c3e3606bfe17d72d936fa0552c90417

File: tests/integration_tests/__init__.py [hunk 9]
```diff
@@ -1,7 +1,6 @@
 # PACKAGE IMPORTS FOR EXTERNAL USAGE
 
-from tests.integration_tests.s3_dataset_test import TestDataset
 from tests.integration_tests.ema_train_integration_test import EMAIntegrationTest
 from tests.integration_tests.lr_test import LRTest
 
-_all__ = [TestDataset, EMAIntegrationTest, LRTest]
+_all__ = [EMAIntegrationTest, LRTest]
```
[the remark this answers is not in the document]
Comment (shaydeci) [reply tests/integration_tests/__init__.py]: Changed in https://github.com/Deci-AI/super-gradients/commit/305ecfd20c3e3606bfe17d72d936fa0552c90417

File: tests/integration_tests/conversion_callback_test.py [hunk 10]
```diff
@@ -5,18 +5,17 @@
 from super_gradients.training import models
 
 from super_gradients import (
-    Trainer,
-    ClassificationTestDatasetInterface,
-    SegmentationTestDatasetInterface,
+    Trainer
```
[the remark this answers is not in the document]
Comment (shaydeci) [reply tests/integration_tests/conversion_callback_test.py]: Changed in https://github.com/Deci-AI/super-gradients/commit/305ecfd20c3e3606bfe17d72d936fa0552c90417

File: src/super_gradients/recipes/dataset_params/cifar10_dataset_params.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (ofrimasad) [reply src/super_gradients/recipes/dataset_params/cifar10_dataset_params.yaml:7]: ok. i see. leave it like this

**Review by ofrimasad — APPROVED** [review]
LGTM
